### PR TITLE
NO-JIRA: drop openshift-sdn references

### PR DESCRIPTION
**Pull request by danwinship**
mostly just cleanup
(there's also a bug in test-kubernetes-e2e.sh but it doesn't matter that much)

### CODE REVIEW

File: openshift-hack/e2e/annotate/rules.go [hunk 1]
```diff
@@ -387,10 +376,6 @@ var (
 			`\[Feature:StorageProvider\]`,
 		},
 
-		// tests that don't pass under openshift-sdn multitenant mode
-		"[Skipped:Network/OpenShiftSDN/Multitenant]": {
-			`\[Feature:NetworkPolicy\]`, // not compatible with multitenant mode
-		},
 		// tests that don't pass under OVN Kubernetes
 		"[Skipped:Network/OVNKubernetes]": {
 			// ovn-kubernetes does not support named ports
```
Comment (atiratree) [inline openshift-hack/e2e/annotate/rules.go:381]: should the following skip be also removed?

https://github.com/openshift/kubernetes/blob/81c1851c0291160c33b40bc22f031293ef756879/openshift-hack/e2e/annotate/rules.go#L401-L403

**Review by atiratree — COMMENTED** [review]
lgtm overall

File: openshift-hack/e2e/annotate/rules.go (hunk 1 above)
Comment (danwinship) [reply openshift-hack/e2e/annotate/rules.go:381]: AFAIK `ibmroks` is still using Calico so they get their own separate set of skips.

(Well, chasing that bugzilla reference leads to OCPBUGS-8771 which claims that the bug is fixed in Calico now, so maybe this skip can be removed, but it's not related to openshift-sdn anyway.)

Comment (atiratree) [reply openshift-hack/e2e/annotate/rules.go:381]: I see, I will leave it up to you if you want to pursue that as a followup

**Review by bertinatto — COMMENTED** [review]
/approve

Thank you so much for doing these cleanups, @danwinship.

**Pull request merged**